### ISSUE TITLE
ggml: create a complete document

### ISSUE THREAD

**Issue by Cyberhan123** [ggerganov/ggml#568]
I am a newbie in C++ and ggml. I want to try to make a ggml version of the [vallex](https://github.com/Plachtaa/VALL-E-X/) model. Currently, I have not found the documentation of ggml. I am not confident just by using the example of the language model.

**Comment by Cyberhan123** [comment]
I would like to try writing documentation.

**Comment by Cyberhan123** [comment]
I checked the source code and it feels like it’s a lot of work and it may take a long time to complete, but I will finish it as soon as possible.
 https://github.com/ggerganov/ggml/pull/571

**Comment by ggerganov** [comment]
Hey, thank you for the interest in the project. If it helps you, please feel free to write docs and examples.

But keep in mind that this work is unlikely to get merged in the repo. `ggml` is changing quickly and it would be better for people more deeply involved in the project to write the docs at some later stage in the future

**Comment by Cyberhan123** [comment]
> Hey, thank you for the interest in the project. If it helps you, please feel free to write docs and examples.
> 
> But keep in mind that this work is unlikely to get merged in the repo. `ggml` is changing quickly and it would be better for people more deeply involved in the project to write the docs at some later stage in the future

You are right, thank you for the great work, when I sorted out the documentation I found that it is not a user friendly library at the moment, but I look forward to its transformation in the future

**Comment by StudyingLover** [comment]
I am learning the use of ggml and recording my learning process, trying to use it as a learning tutorial for novices.url is https://github.com/StudyingLover/ggml-tutorial. If you write a complete, good introductory document, please let me know.
